src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -849,4 +917,40 @@ void DnsMetricsManager::process_dnstap(const dnstap::Dnstap &payload, bool filte
     // process in the "live" bucket. this will parse the resources if we are deep sampling
     live_bucket()->process_dnstap(_deep_sampling_now, payload);
 }
+void DnsMetricsManager::process_cache_dnstap(CacheDnsHandler *cache, bool filtered)
+{
+    if (filtered) {
+        return process_filtered(cache->timestamp);
+    }
+
+    new_event(cache->timestamp);
+    if (cache->payload.getHeaderLen() != 0) {
+        live_bucket()->process_dns_layer(_deep_sampling_now, cache->payload, cache->l3, cache->l4, cache->port, false);
+    } else {
+        live_bucket()->process_dns_layer(cache->l3, cache->l4, cache->side, cache->port, false);
+    }
+}
+
+void DnsMetricsManager::process_cache_pcap(CacheDnsHandler *cache, PacketDirection dir, uint32_t flowkey)
+{
+    new_event(cache->timestamp);
+    if (cache->payload.getHeaderLen() == 0) {
+        live_bucket()->process_dns_layer(cache->l3, cache->l4, cache->side, cache->port, false);
+
+    } else {
+        live_bucket()->process_dns_layer(_deep_sampling_now, cache->payload, cache->l3, cache->l4, cache->port, false);
+
+        if (group_enabled(group::DnsMetrics::DnsTransactions)) {
```

Comment:
DNS is not caching transaction information. should we figure out a way to do that?